### PR TITLE
[C++ verification] Standardize the cpp new/delete

### PR DESCRIPTION
**Pull request by XLiZHI**
see #825

### CODE REVIEW

File: src/goto-symex/builtin_functions.cpp [hunk 1]
```diff
@@ -588,27 +588,33 @@ void goto_symext::symex_cpp_new(const expr2tc &lhs, const sideeffect2t &code)
...
+  expr2tc ptr_obj = pointer_object2tc(pointer_type2(), ptr_rhs);
+  track_new_pointer(ptr_obj, newtype, size);
 
   dynamic_memory.emplace_back(
     rhs_copy, cur_state->guard, false, symbol.name.as_string());
 }
 
-// XXX - implement as a call to free?
-void goto_symext::symex_cpp_delete(const expr2tc &)
+void goto_symext::symex_cpp_delete(const expr2tc &expr)
```
Comment (fbrausse) [inline src/goto-symex/builtin_functions.cpp]: Why not implement it as a call to symex_free()?

Comment (XLiZHI) [reply src/goto-symex/builtin_functions.cpp]: Hi @fbrausse, Do you mean to use `symex_free()` inside this function? 

Maybe we don't need `symex_cpp_delete`, I tried using symex_free instead:

```
  else if (is_code_cpp_del_array2t(code2) || is_code_cpp_delete2t(code2))
  {
    replace_dynamic_allocation(code2);
    replace_nondet(code2);

    symex_free(code2);
  }
```

Comment (fbrausse) [reply src/goto-symex/builtin_functions.cpp]: We should probably check that the allocation has actually been performed with `new[]` for `delete[]`, but other than that I think symex_cpp_delete() can just delegate to symex_free().

**Review by lucasccordeiro — APPROVED** [review]
This is a very nice PR!

Thanks for submitting it, @XLiZHI.

**Review by rafaelsamenezes — APPROVED** [review]

**Pull request merged**